solr/core/src/java/org/apache/solr/search/neural/KnnQParser.java

```diff
@@ -84,30 +84,20 @@ public Query parse() {
   }
 
   private Query getFilterQuery() throws SolrException {
-    if (!isFilter()) {
+    boolean isSubQuery = recurseCount != 0;
+    if (!isFilter() && !isSubQuery) {
       String[] filterQueries = req.getParams().getParams(CommonParams.FQ);
       if (filterQueries != null && filterQueries.length != 0) {
-        List<Query> filters;
-
         try {
-          filters = QueryUtils.parseFilterQueries(req, true);
-        } catch (SyntaxError e) {
+          List<Query> filters = QueryUtils.parseFilterQueries(req, true);
+          SolrIndexSearcher.ProcessedFilter processedFilter =
+              req.getSearcher().getProcessedFilter(filters);
+          return processedFilter.filter;
+        } catch (SyntaxError | IOException e) {
```

Comment:
Don't catch SyntaxError; propagate it. getFilterQuery will indirectly parse and this is in the context of query parsing; SyntaxError is a declared exception of QueryParser.parse.

Reply:
Done, didn't notice in my original review of the code for 9.1 !
Thanks for pointing this out!